### PR TITLE
Edit of Artifact Store page

### PR DESCRIPTION
**Pull request by sascala**

### CODE REVIEW

File: docs/docs/artifact-store.md [hunk 1]
```diff
@@ -4,39 +4,38 @@ title: Artifact Store
...
-In Marathon we call this place the artifact store.
+Deployments inside a distributed system need a location where application-specific resources can be found. In Marathon, we call this place the artifact store.
 
 
 ## Usage
 
-The use of the Marathon Artifact Store functionality is completely optional.
-All functionality and goals can be accomplished manually as well.
-Marathon tries to simplify several use cases.
+Use of the Marathon artifact store is completely optional: you can create all of the artifact atore functionality manually as well. The Marathon artifact store simplifies several use cases.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: What are some example use cases?

File: docs/docs/artifact-store.md [hunk 2]
```diff
@@ -4,39 +4,38 @@ title: Artifact Store
...
 
 
 ## Artifact Store Backend
 
-Marathon supports different storage system, that can be used as artifact store.
-The type of the artifact store is configured via the command line.
-Example: start with --artifact_store hdfs://localhost:54310/path/to/store to use Hadoop DFS 
-as artifact storage backend.
+Marathon supports several different storage system backends for the artifact store.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: Where can we find out which ones are supported?

File: docs/docs/artifact-store.md [hunk 3]
```diff
@@ -4,39 +4,38 @@ title: Artifact Store
...
 
-### Upload an artifact to the artifact store
+## Artifact REST Endpoint
 
-Upload an artifact to the artifact store.
-A multipart form upload request has to be performed.
-The form parameter name has to be ```file```.
-The filename used in the artifact store, is the same as given by the form parameter.
-The response holds the URL of the artifact in the artifact store in the Location Header.
+A special endpoint enables you to access and manipulate the artifacts in the artifact store. The URLs of the created artifacts can be used as URIs of an application definition.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: I'd like to put an example here.

File: docs/docs/artifact-store.md [hunk 4]
```diff
@@ -4,39 +4,38 @@ title: Artifact Store
...
+
+See the [REST API documentation](http://mesosphere.github.io/marathon/docs/generated/api.html#v2_artifacts) for more information.
+
+### Upload an Artifact to the Artifact Store
+
+To upload an artifact to the artifact stopre, you must perform a multipart form upload request.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: Would it be more precise to say that your *application* "must perform a multipart form upload request"?

File: docs/docs/artifact-store.md [hunk 5]
```diff
@@ -150,25 +143,29 @@ Server: Jetty(8.1.11.v20130520)
 ```
 
 
-## Automatic Artifact Storing
+## Automatic Artifact Storage
+
+An AppDefinition holds a sequence of URIs that Marathon fetches when it starts each app instance.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: The sentence doesn't seem important here.

File: docs/docs/artifact-store.md [hunk 6]
```diff
@@ -150,25 +143,29 @@ Server: Jetty(8.1.11.v20130520)
...
 
-An AppDefinition holds a sequence of URIs, that get fetched on each instance, that gets started.
 The artifact could be fetched directly from the source, or put into the artifact store.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: The sentence doesn't seem important here.

File: docs/docs/artifact-store.md [hunk 7]
```diff
@@ -150,25 +143,29 @@ Server: Jetty(8.1.11.v20130520)
...
 
-The AppDefinition has a field storeUrls, which holds an array of URL strings.
+You can automatically store artifacts in the artifact store with the ```storeURLs``` field of the AppDefinition. The ```storeURLs``` field holds an array of URL strings.
 Every URL here is processed in this way:
 
-* The URL gets downloaded
-* The byte stream is stored in the asset store
-* The asset store url is added to the AppDefinition uris list
-* The url is removed from the storeUrls array
+* The URL is downloaded.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: What downloads it? Where is it downloaded to?

File: docs/docs/artifact-store.md [hunk 8]
```diff
@@ -150,25 +143,29 @@ Server: Jetty(8.1.11.v20130520)
...
-The AppDefinition has a field storeUrls, which holds an array of URL strings.
+You can automatically store artifacts in the artifact store with the ```storeURLs``` field of the AppDefinition. The ```storeURLs``` field holds an array of URL strings.
 Every URL here is processed in this way:
 
-* The URL gets downloaded
-* The byte stream is stored in the asset store
-* The asset store url is added to the AppDefinition uris list
-* The url is removed from the storeUrls array
+* The URL is downloaded. 
+* The byte stream is stored in the asset store.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: For my edification: the byte stream of what? the URL? the artifact?

File: docs/docs/artifact-store.md [hunk 9]
```diff
@@ -150,25 +143,29 @@ Server: Jetty(8.1.11.v20130520)
...
+* Marathon adds the asset store URL to the AppDefinition URIs list.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: I'm just guessing that it's Marathon that is adding the asset store URL here.

File: docs/docs/artifact-store.md [hunk 10]
```diff
@@ -150,25 +143,29 @@ Server: Jetty(8.1.11.v20130520)
...
+* Marathon removes the URL from the storeUrls array.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: Same as above.

File: docs/docs/artifact-store.md [hunk 11]
```diff
@@ -150,25 +143,29 @@ Server: Jetty(8.1.11.v20130520)
...
 
-As a result, all storeUrls are accessible from the artifact store.
-All instances that will run the application, will load the needed assets from the artifact store.
+As a result, all storeUrls are accessible from the artifact store. All instances that run the application are able to load the necessary assets from the artifact store.
 
+### Prerequisites
+
+To use this feature, all assets need to be resolvable by marathon itself.
+The HTTP server should support HTTP ETag Header in order to circumvent manual content hash creation.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: Don't we discuss below doing just that without the ETag header?

File: docs/docs/artifact-store.md [hunk 12]
```diff
@@ -150,25 +143,29 @@ Server: Jetty(8.1.11.v20130520)
...
 
-### Create an application definition with automatic artifact resolution
+### Create an Application Definition with Automatic Artifact Resolution
```
Comment (sascala) [inline docs/docs/artifact-store.md]: I'd like to say a little about what automatic artifact resolution is.

File: docs/docs/artifact-store.md [hunk 13]
```diff
@@ -250,22 +247,17 @@ Transfer-Encoding: chunked
...
-* if ETag is not available (dumb HTTP server), download the resource and compute the Sha-1 hash manually
-* filename of the url remains the same
+The path in the asset store is computed in the following way:
 
-The complete path is {artifact store base}/{ETag or ContentHash}/{filename of asset}
-This effectively will create a path, that is unique to the content of the resource.
-If the same URL holds a different entity, a new path is created.
-The same path is downloaded only once. 
-In other words if the path is available in the artifact store, it is not downloaded again.
+1. HEAD Request to the asset.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: I'd like to be more precise/detailed here.

File: docs/docs/artifact-store.md [hunk 14]
```diff
@@ -250,22 +247,17 @@ Transfer-Encoding: chunked
...
+2. If ETag is available, take ETag HTTP header.
+3. If ETag is not available, download the resource and compute the Sha-1 hash manually.
+4. The filename of the url remains the same.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: Is this a part of the process? What does this mean?

File: docs/docs/artifact-store.md [hunk 15]
```diff
@@ -250,22 +247,17 @@ Transfer-Encoding: chunked
...
 
+The complete path to the artifact is ```{artifact store base}/{ETag or ContentHash}/{filename of asset}```.
 
-### Prerequisites
+Marathon creates a path that is unique to the content of the resource.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: I'm just guessing here at what creates the path

File: docs/docs/artifact-store.md [hunk 16]
```diff
@@ -250,22 +247,17 @@ Transfer-Encoding: chunked
...
 
-To use this feature, all assets need to be resolvable by marathon itself.
-To circumvent manual content hash creation, the http server should support HTTP ETag Header. 
+If the same URL holds a different entity, a new path is created.
```
Comment (sascala) [inline docs/docs/artifact-store.md]: What creates the new path?

File: docs/docs/artifact-store.md (hunk 1 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: s/atore/store

Comment (aquamatthias) [inline docs/docs/artifact-store.md]: The main use case: artifacts needed for a deployment have to be available in the cluster. The artifact store could be that place.

File: docs/docs/artifact-store.md (hunk 2 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: run Marathon --help and see the options in  --artifact_store. Currently supported:
```
hdfs, file
```

File: docs/docs/artifact-store.md (hunk 3 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: Please see the rest api documentation for an example.

File: docs/docs/artifact-store.md (hunk 4 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: No. That step has to be performed, before the application is created, that uses those artifacts.

File: docs/docs/artifact-store.md (hunk 6 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: That sentence tries to explain the difference between `uris` (which BTW is deprecated and replaced by `fetch`) and storeURLSs. As long as the difference becomes clear, the sentence can be removed.

File: docs/docs/artifact-store.md (hunk 7 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: the artifact is downloaded in to the artifact store.

File: docs/docs/artifact-store.md (hunk 8 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: the contents of the url.

File: docs/docs/artifact-store.md (hunk 9 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: Yes: the original url (where this asset is downloaded from) is replaced by the store url.

File: docs/docs/artifact-store.md (hunk 10 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: After creating an app, all storeURLs are replaced and put into uris.

File: docs/docs/artifact-store.md (hunk 11 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: Marathon handles the absence of the ETag functionality, but we strongly encourage a setup that has this functionality enabled.

File: docs/docs/artifact-store.md (hunk 12 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: Automatic here means: do not use the /v2/artifacts API, but the storeURLs in the app definition (which handles all the details)

File: docs/docs/artifact-store.md (hunk 13 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: See https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html Paragraph 9.4 for HEAD requests.

File: docs/docs/artifact-store.md (hunk 14 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: Each URL points to a file. Since the URL is replaced with a URI, we make sure, the filename stays the same.

File: docs/docs/artifact-store.md (hunk 15 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: since the content hash is used in the path, the path is unique.

File: docs/docs/artifact-store.md (hunk 16 above)
Comment (aquamatthias) [inline docs/docs/artifact-store.md]: The content hash.

**Comment by aquamatthias** [comment]
@sascala We want to deprecate the artifact store. Suggestion: we put a warning on top of the documentation, that signals this. What do you think?

**Comment by aquamatthias** [comment]
Close for now - please reopen when ready.

**Comment by luhkevin** [comment]
Is the artifact store still planned for deprecation? What's the motivation behind deprecating it?

**Comment by aquamatthias** [comment]
@luhkevin the artifact store takes artifacts and puts those into the storage backend. It is not a core functionality of Marathon and can be solved outside of Marathon easily.
